src/group_theory/group_action/group.lean

```diff
@@ -23,29 +23,37 @@ section mul_action
 section units
 variables [monoid α] [mul_action α β]
 
-@[simp] lemma units.inv_smul_smul (u : units α) (x : β) :
+@[simp, to_additive] lemma units.inv_smul_smul (u : units α) (x : β) :
   (↑u⁻¹:α) • (u:α) • x = x :=
 by rw [smul_smul, u.inv_mul, one_smul]
 
-@[simp] lemma units.smul_inv_smul (u : units α) (x : β) :
+@[simp, to_additive] lemma units.smul_inv_smul (u : units α) (x : β) :
   (u:α) • (↑u⁻¹:α) • x = x :=
 by rw [smul_smul, u.mul_inv, one_smul]
 
+/-- If a monoid `α` acts on `β`, then each `u : units α` defines a permutation of `β`. -/
+@[to_additive] def units.smul_perm (u : units α) : equiv.perm β :=
+⟨λ x, (u:α) • x, λ x, (↑u⁻¹:α) • x, u.inv_smul_smul, u.smul_inv_smul⟩
+
+/-- If an additive monoid `α` acts on `β`, then each `u : add_units α` defines a permutation
+of `β`. -/
+add_decl_doc add_units.vadd_perm
+
 /-- If a monoid `α` acts on `β`, then each `u : units α` defines a permutation of `β`. -/
 def units.smul_perm_hom : units α →* equiv.perm β :=
-{ to_fun := λ u, ⟨λ x, (u:α) • x, λ x, (↑u⁻¹:α) • x, u.inv_smul_smul, u.smul_inv_smul⟩,
+{ to_fun := units.smul_perm,
   map_one' := equiv.ext $ one_smul α,
   map_mul' := λ u₁ u₂, equiv.ext $ mul_smul (u₁:α) u₂ }
```

Comment:
Can you add the additive version of this as `add_units α →+ additive (equiv.perm β)`?

Reply:
Done.